src/flamenco/runtime/fd_blockstore.c

```diff
@@ -190,6 +193,52 @@ static inline void check_read_write_err( int err ) {
   }
 }
 
+bool 
+fd_blockstore_invalid_archive( fd_blockstore_t * blockstore, fd_blockstore_archive_metadata_t * metadata ) {
+  return metadata->sequence_num != 0 
+         || ( metadata->lrw_off < FD_BLOCKSTORE_ARCHIVE_START ) 
+         || ( metadata->mrw_end < FD_BLOCKSTORE_ARCHIVE_START )
+         || ( metadata->fd_size_max != blockstore->fd_size_max ); // should be initialized same as archive file
+}
+
+#define check_read_err_safe( cond, msg )               \
+  if( FD_UNLIKELY( cond ) ) {                     \
+    FD_LOG_WARNING(( "[%s] %s", __func__, msg )); \
+    return FD_BLOCKSTORE_ERR_SLOT_MISSING;        \
+  }
```

Comment:
kind of a funky macro here to pass in the `msg` like that

you can just use string concatenation `##` c preprocessor syntax and so `[%s] ## msg` that